tflint/config.go

```diff
@@ -212,7 +245,17 @@ func loadConfig(file afero.File) (*Config, error) {
 	}
 
 	parser := hclparse.NewParser()
-	f, diags := parser.ParseHCL(src, file.Name())
+	var f *hcl.File
+	var diags hcl.Diagnostics
+
+	// Parse based on file extension
+	switch {
+	case strings.HasSuffix(strings.ToLower(file.Name()), ".json"):
+		f, diags = parser.ParseJSON(src, file.Name())
+	default:
+		f, diags = parser.ParseHCL(src, file.Name())
+	}
```

Comment:
A switch expression with case expressions is easier to read than a bare switch:

```suggestion
	switch filepath.Ext(file.Name()) {
	case ".json":
		f, diags = parser.ParseJSON(src, file.Name())
	default:
		f, diags = parser.ParseHCL(src, file.Name())
	}
```

You'll need to import `filepath` too